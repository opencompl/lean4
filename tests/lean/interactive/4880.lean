/-!
# Ensure autoparam errors are placed at elaboration position

Before, errors were placed at the beginning of the file.
-/


/-!
Testing `infer_instance`, which defers a typeclass problem beyond the tactic script execution.
-/
class A

-- For structure instance elaboration ...
structure B where
  h1 : A := by infer_instance

example : B where
          --^ collectDiagnostics

-- ... and for app elaboration.
def baz (_h1 : A := by infer_instance) : Nat := 1

example : Nat := baz
               --^ collectDiagnostics


/-!
Testing a tactic that immediately throws an error, but incrementality resets the ref
from the syntax for the tactic (which would be a `.missing` position for autoparams).
-/

structure B' where
  h1 : A := by done

example : B' where
           --^ collectDiagnostics
