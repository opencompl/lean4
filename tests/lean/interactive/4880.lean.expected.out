{"version": 1,
 "uri": "file:///4880.lean",
 "diagnostics":
 [{"source": "Lean 4",
   "severity": 1,
   "range":
   {"start": {"line": 16, "character": 12},
    "end": {"line": 16, "character": 17}},
   "message":
   "failed to synthesize\n  A\nAdditional diagnostic information may be available using the `set_option diagnostics true` command.",
   "fullRange":
   {"start": {"line": 16, "character": 12},
    "end": {"line": 16, "character": 17}}},
  {"source": "Lean 4",
   "severity": 1,
   "range":
   {"start": {"line": 22, "character": 17},
    "end": {"line": 22, "character": 20}},
   "message":
   "failed to synthesize\n  A\nAdditional diagnostic information may be available using the `set_option diagnostics true` command.",
   "fullRange":
   {"start": {"line": 22, "character": 17},
    "end": {"line": 22, "character": 20}}},
  {"source": "Lean 4",
   "severity": 1,
   "range":
   {"start": {"line": 34, "character": 13},
    "end": {"line": 34, "character": 18}},
   "message": "unsolved goals\n‚ä¢ A",
   "fullRange":
   {"start": {"line": 34, "character": 13},
    "end": {"line": 34, "character": 18}}}]}
